enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx

```diff
@@ -64,12 +71,16 @@ const QuestionEditorInner = () => {
         </Tabs.Panel>
       </Tabs>
 
-      {isSaveEnabled && isSaveFormOpen && (
-        <Overlay center>
-          <Paper>
-            <InteractiveQuestion.SaveQuestionForm onClose={closeSaveForm} />
-          </Paper>
-        </Overlay>
+      {/* Refer to the SaveQuestionProvider for context on why we have to do it like this */}
```

Comment:
I read the comment but I still haven't understood what we're doing, why, and what was the problem 😅

Reply:
I've changed it a little bit so hopefully it's a little better explained

Reply:
I still don't understand what we mean by "we have to do it like this" 😅
When a modal uses an entity, i think it's fairly normal to only render it when needed instead of using `open={flag}`